samples/ModbusMaster/component.mk

```diff
@@ -0,0 +1,40 @@
+#### overridable rBoot options ####
+
+## use rboot build mode
+RBOOT_ENABLED		?= 1
+
+## enable big flash support (for multiple roms, each in separate 1mb block of flash)
+RBOOT_BIG_FLASH		?= 1
+
+## two rom mode (where two roms sit in the same 1mb block of flash)
+#RBOOT_TWO_ROMS		?= 1
+
+## size of the flash chip
+SPI_SIZE			?= 4M
+
+## output file for first rom (.bin will be appended)
+#RBOOT_ROM_0 		?= rom0
+
+## input linker file for first rom
+#RBOOT_LD_0			?= rom0.ld
+
+## these next options only needed when using two rom mode
+#RBOOT_ROM_1		?= rom1
+#RBOOT_LD_1			?= rom1.ld
+
+## size of the spiffs to create
+SPIFF_SIZE			?= 65536
+
+## option to completely disable spiffs
+#DISABLE_SPIFFS		?= 1
+
+## flash offsets for spiffs, set if using two rom mode or not on a 4mb flash
+## (spiffs location defaults to the mb after the rom slot on 4mb flash)
+#RBOOT_SPIFFS_0		?= 0x100000
+#RBOOT_SPIFFS_1		?= 0x300000
+
+#ENABLE_SMART_CONFIG = 1
+ARDUINO_LIBRARIES := ModbusMaster
+
+COMPONENT_SRCDIRS = app
+COMPONENT_INCDIRS = include
```

Comment:
Also there is no `include` direcory, this should be `src` (Sorry, forgot to mention)

Reply:
I added the relative path, however I suspect it isn't the proper way?

Reply:
Sorry, I'm talking about the wrong `component.mk` here! There needs to be a `component.mk` file for the `ModbusMaster` library. Because it customises the path information this has to be visible to Sming _before_ the library is pulled in, so you have two choices for this:

1. Move the `ModbusMaster` submodule into a sub-directory:

```
|_  Sming/
    |_ Libraries/
        |_ ModbusMaster/
            |_ ModbusMaster/   # The submodule
            component.mk
            README.rst # Add this yourself
```

This way, you can more easily add customisations specific to Sming without having to update the submodule. Moving is fairly straightforward:

```
Sming/Libraries$ git mv ModbusMaster tmp
Sming/Libraries$ mkdir ModbusMaster
Sming/Libraries$ git mv tmp ModbusMaster/ModbusMaster
```

The `component.mk` file will contain these lines:

```
COMPONENT_SRCDIRS := ModbusMaster/src
COMPONENT_INCDIRS := ModbusMaster/src
```

This approach also allows you to add your own README as an introduction page to the library. This can contain any advice or tips specific to Sming. You can see how this works for the `ArduinoJson6` library - https://sming.readthedocs.io/en/latest/_inc/Sming/Libraries/ArduinoJson6/index.html. The `README.rst` file is very simple - click on the `Edit on github` link to see the contents.


2. Alternatively, just add the `component.mk` file as a patch:

```
|_  Sming/
    |_ Libraries/
        |_ .patches/
            |_ ModbusMaster/
                |_ component.mk
```

The component.mk would look like this:

```
COMPONENT_SRCDIRS := src
COMPONENT_INCDIRS := src
```

Reply:
The notes you've added to the sample application would be more appropriate in the library README.